src/source/ObjcGenerator.scala

```diff
@@ -235,7 +235,7 @@ class ObjcGenerator(spec: Spec) extends BaseObjcGenerator(spec) {
         writeAlignedObjcCall(w, decl, r.fields, "", f => (idObjc.field(f.ident), s"(${marshal.paramType(f.ty)})${idObjc.local(f.ident)}"))
         w.wl
         w.braced {
-          val call = s"return [[self alloc] init$firstInitializerArg"
+          val call = s"return [($self*)[self alloc] init$firstInitializerArg"
```

Comment:
Would a cast to instancetype make more sense here?  Looking at the generated code it seems inconsistent to use the explict type here but `instancetype` in the return type declaration 2 lines above.  Of course, I'm a bit surprised by the ObjC behavior you're fixing here too, so I'm not sure if it's more quirky than I expect.

Reply:
To be honest, I have only worked on Objective C code a handful of times so I may be off here, but my understanding of the instancetype keyword is that it can only be used as a return type in function signatures, and casting to it is not possible.  Doing [(instancetype)[self alloc] results in a compiler error, as does instancetype* .

Reply:
Huh, I was always a polyglot too so that may be a quirk or Objective C I'm not clear on.  It would be nice if instancetype could be used more broadly, but if it doesn't then the direct cast is the best we can do.